packages/loopring_v3/contracts/aux/bridge/Bridge.sol

```diff
@@ -0,0 +1,902 @@
+            uint packedData;
+            bytes20 dataHash;
+            assembly {
+                header     := calldataload(    txsDataPtr     )
+                packedData := calldataload(add(txsDataPtr, 42))
+                dataHash   := and(calldataload(add(txsDataPtr, 78)), 0xffffffffffffffffffffffffffffffffffffffff000000000000000000000000)
+            }
+            require(
+                // txType == ExchangeData.TransactionType.WITHDRAWAL &&
+                // withdrawal.type == 1 &&
+                header & 0xffff == (uint(ExchangeData.TransactionType.WITHDRAWAL) << 8) | 1 &&
+                // withdrawal.tokenID == token.tokenID &&
+                // withdrawal.amount == token.amount &&
+                // withdrawal.fee == 0,
+                packedData & 0xffffffffffffffffffffffffffff0000ffff == (uint(token.tokenID) << 128) | (token.amount << 32) &&
+                onchainDataHash == dataHash,
+                "INVALID_BRIDGE_WITHDRAWAL_TX_DATA"
+            );
+
+            ctx.txsDataPtr += ExchangeData.TX_DATA_AVAILABILITY_SIZE;
+        }
+    }
+
+    function _storeTransfers(
+        bytes memory transfers,
+        address from
+        )
+        internal
+    {
+        uint batchID = batchIDGenerator++;
+
+        // Store transfers to distribute at a later time
+        bytes32 hash = _hashTransferBatch(transfers);
+        require(pendingTransfers[batchID][hash] == 0, "DUPLICATE_BATCH");
+        pendingTransfers[batchID][hash] = block.timestamp;
+
+        // Log transfers to do
+        emit Transfers(batchID, transfers, from);
+    }
+
+    function _deposit(
+        address from,
+        address token,
+        uint96  amount
+        )
+        internal
+    {
+        if (amount == 0) {
+            return;
+        }
+
+        if (from == address(this) && token != address(0)) {
+            ERC20(token).approve(address(depositContract), amount);
+        }
+        // Do the token transfer directly to the exchange
+        uint ethValue = (token == address(0)) ? amount : 0;
+        exchange.deposit{value: ethValue}(from, address(this), token, amount, new bytes(0));
+    }
+
+    function _connectorCall(
+        Context          memory   ctx,
+        ConnectorCalls   calldata connectorCalls,
+        uint                      n,
+        ConnectorCalls[] calldata allConnectorCalls
+        )
+        internal
+        returns (BridgeTransfer[] memory transfers)
+    {
+        require(connectorCalls.connector != address(this), "INVALID_CONNECTOR");
+        require(trustedConnectors[connectorCalls.connector], "ONLY_TRUSTED_CONNECTORS_SUPPORTED");
+
+        // Check if the minimum amount of gas required is achieved
+        bytes memory txData = _getConnectorCallData(ctx, IBridgeConnector.getMinGasLimit.selector, allConnectorCalls, n);
+        (bool success, bytes memory returnData) = connectorCalls.connector.fastCall(GAS_LIMIT_CHECK_GAS_LIMIT, 0, txData);
```

Comment:
What is this `fastCall` for? The `fastDelegatecall` is the real interaction right?